src/blocks/DistrivialApp.tsx

```diff
@@ -0,0 +1,251 @@
+import { useState, useEffect } from 'react';
+import { ComposerComponentProps, FeedComponentProps } from './types';
+
+interface Question {
+  question: string;
+  correct_answer: string;
+  incorrect_answers: string[];
+}
+
+// Decode HTML entities
+const decodeHTML = (html: string) => {
+  const text = document.createElement('textarea');
+  text.innerHTML = html;
+  return text.value;
+};
+
+// Shuffle array elements
+const shuffleArray = (array: any[]) => {
+  return array.sort(() => Math.random() - 0.5);
+};
+
+// Fetch questions from API
+const fetchQuestions = async (): Promise<Question[]> => {
+  const response = await fetch('https://opentdb.com/api.php?amount=5&type=multiple');
+  const data = await response.json();
+  return data.results.map((item: any) => ({
+    question: decodeHTML(item.question),
+    correct_answer: decodeHTML(item.correct_answer),
+    incorrect_answers: item.incorrect_answers.map((ans: string) => decodeHTML(ans)),
+  }));
+};
+
+// DistrivialFeedComponent definition
+export const DistrivialFeedComponent = ({ model }: FeedComponentProps) => {
+  const questions: any[] = model.questions || [];
```

Comment:
make sure to get the defaults of the state from the model, like so:
```
let questions = model.data['questions'] ? JSON.parse(model.data['questions']) : [];
```
As mentioned below, you should be storing the data inside of the data property of the model, rather than creating new variables just for your miniapp.